### ISSUE TITLE
general questions

### ISSUE THREAD

**Issue by x-077** [gsdenys/amqp-client#8]
Hello, 

I've been playing with LUA with openresty and its great so far. 

I would have few question please:  

- does amqp-client ha ve a 100% nonblocking behavior ?
- could please advise in which context is it best to use ? ( init_by_lua_* , content_by_lua_* ) ?
- is it possible to "use" it in the background ? Meaning that the connection never dies unless the client get disconnected by the amqp server ? 

To give you some context, in my case I would like to use it only as a consumer to create / delete / update keys in a local redis ( using resty.redis ) independently of traffic / user request. 

 Meaning that the amqp client connection should be persistant and keep interacting with the local redis.
 
 Basically, Im wondering if I could replace one of my micro service using nodejs / amqp by openresty / amqp-client. 
 
 Thanks.
